data/json/recipes/practice/skills.json

```
@@ -0,0 +1,47 @@
+[
+  {
+    "id": "prac_computer_beg",
+    "type": "practice",
+    "activity_level": "NO_EXERCISE",
+    "category": "CC_PRACTICE",
+    "subcategory": "CSC_PRACTICE_SKILLS",
+    "name": "computer (beginner)",
+    "description": "Practice the basics of using a computer",
+    "skill_used": "computer",
+    "time": "30 m",
+    "practice_data": { "min_difficulty": 0, "max_difficulty": 1, "skill_limit": 2 },
+    "autolearn": [ [ "computer", 1 ] ],
+    "book_learn": [ [ "mag_computer", 0 ], [ "howto_computer", 0 ] ],
+    "tools": [ [ [ "laptop", 25 ], [ "eink_tablet_pc", 10 ], [ "smart_phone", 10 ] ] ]
+  },
+  {
+    "id": "prac_computer_int",
+    "type": "practice",
+    "activity_level": "NO_EXERCISE",
+    "category": "CC_PRACTICE",
+    "subcategory": "CSC_PRACTICE_SKILLS",
+    "name": "computer (intermediate)",
+    "description": "Practice using the command line, writing and running scripts making use of common algorithms and data structures",
+    "skill_used": "computer",
+    "time": "1 h",
+    "practice_data": { "min_difficulty": 2, "max_difficulty": 4, "skill_limit": 5 },
+    "autolearn": [ [ "computer", 4 ] ],
+    "book_learn": [ [ "manual_computers", 2 ], [ "computer_science", 3 ], [ "textbook_computer", 3 ] ],
+    "tools": [ [ [ "laptop", 50 ] ] ]
+  },
+  {
+    "id": "prac_computer_adv",
+    "type": "practice",
+    "activity_level": "NO_EXERCISE",
+    "category": "CC_PRACTICE",
+    "subcategory": "CSC_PRACTICE_SKILLS",
+    "name": "computer (advanced)",
+    "description": "Practice coding advanced algorithms and software exploits",
+    "skill_used": "computer",
+    "time": "90 m",
+    "practice_data": { "min_difficulty": 5, "max_difficulty": 7 },
+    "autolearn": [ [ "computer", 7 ] ],
+    "book_learn": [ [ "SICP", 6 ] ],
+    "tools": [ [ [ "laptop", 75 ] ] ]
```

```suggestion
    "tools": [ [ [ "laptop", 50 ] ], [ [ "electrohack", 50 ] ] ]
```
I'm not sure an electrohack is specifically the thing here, but I think we should add something to make it clear why this is advanced and why we don't just extend the medium one to go to a higher level.

---

electrohack doesn't involve computer skill, but I could add in the hackPRO software as an extra requirement if we want to focus this practice on hacking

The original idea was just that you'd be writing more complex software at this level, but that probably wouldn't help much with hacking.

---

Hmm. Having software as a required tool in general is a good idea. I'd say that for now since computing at high levels is all hacking we could include that. Later on we might specifically see a hacking proficiency and something else for programming other stuff if it ever becomes relevant.